src/solver/optimisation/opt_rename_problem.cpp

```diff
@@ -0,0 +1,364 @@
+                                        const std::string& destination)
+{
+    origin_ = origin;
+    destination_ = destination;
+    SetLinkVariableName(var, "IntercoIndirectCost");
+}
+
+void VariableNamer::SetShortTermStorageVariableName(unsigned int var,
+                                                    const std::string& variableType,
+                                                    const std::string& shortTermStorageName)
+{
+    const auto location = LocationIdentifier(area_, AREA) + SEPARATOR + "ShortTermStorage" + "<"
+                          + shortTermStorageName + ">";
+    targetUpdater_.UpdateTargetAtIndex(
+      BuildName(variableType, location, TimeIdentifier(timeStep_, HOUR)), var);
+}
+
+void VariableNamer::ShortTermStorageInjection(unsigned int var, const std::string& shortTermStorageName)
+{
+    SetShortTermStorageVariableName(var, "Injection", shortTermStorageName);
+}
+
+void VariableNamer::ShortTermStorageWithdrawal(unsigned int var, const std::string& shortTermStorageName)
+{
+    SetShortTermStorageVariableName(var, "Withdrawal", shortTermStorageName);
+}
+
+void VariableNamer::ShortTermStorageLevel(unsigned int var, const std::string& shortTermStorageName)
+{
+    SetShortTermStorageVariableName(var, "Level", shortTermStorageName);
+}
+
+void VariableNamer::HydProd(unsigned int var)
+{
+    SetAreaVariableName(var, "HydProd");
+}
+
+void VariableNamer::HydProdDown(unsigned int var)
+{
+    SetAreaVariableName(var, "HydProdDown");
+}
+
+void VariableNamer::HydProdUp(unsigned int var)
+{
+    SetAreaVariableName(var, "HydProdUp");
+}
+
+void VariableNamer::Pumping(unsigned int var)
+{
+    SetAreaVariableName(var, "Pumping");
+}
+
+void VariableNamer::HydroLevel(unsigned int var)
+{
+    SetAreaVariableName(var, "HydroLevel");
+}
+
+void VariableNamer::Overflow(unsigned int var)
+{
+    SetAreaVariableName(var, "Overflow");
+}
+
+void VariableNamer::LayerStorage(unsigned int var, int layerIndex)
+{
+    SetAreaVariableName(var, "LayerStorage", layerIndex);
+}
+
+void VariableNamer::FinalStorage(unsigned int var)
+{
+    SetAreaVariableName(var, "FinalStorage");
+}
+
+void VariableNamer::PositiveUnsuppliedEnergy(unsigned int var)
+{
+    SetAreaVariableName(var, "PositiveUnsuppliedEnergy");
+}
+
+void VariableNamer::NegativeUnsuppliedEnergy(unsigned int var)
+{
+    SetAreaVariableName(var, "NegativeUnsuppliedEnergy");
+}
+
+void VariableNamer::AreaBalance(unsigned int var)
+{
+    SetAreaVariableName(var, "AreaBalance");
+}
+
+void ConstraintNamer::FlowDissociation(unsigned int numConstraint,
+                                       const std::string& origin,
+                                       const std::string& destination)
+{
+    targetUpdater_.UpdateTargetAtIndex(
+      BuildName("FlowDissociation",
+                LocationIdentifier(origin + AREA_SEP + destination, LINK),
+                TimeIdentifier(timeStep_, HOUR)),
+      numConstraint);
```

Comment:
Since this constraint seems to be on the link, why not use similar mechanisms as in `VariableNamer` ? Maybe you could factor some stuff since `SetAreaVariableName` & `SetAreaConstraintName` would share a lot